src/FilesystemAdapter.php

```diff
@@ -101,7 +101,9 @@ public function get(string $key): PromiseInterface
                 Stream::close($file);
             }
 
-            return new FileObject($key, new PsrStream($copy));
+            return new FileObject($key, new PsrStream($copy), [
+                'ContentType' => mime_content_type($path),
```

Comment:
`mime_content_type` returns `false` upon failure. We might want to default to `null` in such case:

```suggestion
                'ContentType' => mime_content_type($path) ?: null,
```